### PR TITLE
Remove (almost all) panickers from trie module

### PR DESCRIPTION
**Pull request by rphmeier**
Those in the trie iterator still remain. Not sure if it's worth removing them since it would mean making the iterator item a `Result<TrieItem>`.

Trie errors will still immediately panic (probably better to fail fast than limp along in a broken state), but the infrastructure is in place to catch them and ask for trie nodes from the network.

### CODE REVIEW

**Comment by gavofyork** [comment]
tests failing...

**Comment by rphmeier** [comment]
appveyor failure is spurious

File: ethcore/src/account.rs [hunk 1]
```diff
@@ -136,7 +138,11 @@ impl Account {
 				SecTrieDBMut would not set it to an invalid state root. Therefore the root is valid and DB creation \
 				using it will not fail.");
 
-			(Filth::Clean, H256::from(db.get(key).map_or(U256::zero(), |v| -> U256 {decode(v)})))
+			let item: U256 = match db.get(key){
+				Ok(x) => x.map_or_else(|| U256::zero(), decode),
```
Comment (gavofyork) [inline ethcore/src/account.rs]: `x.map_or_else(U256::zero, decode)`?

File: ethcore/src/state.rs [hunk 2]
```diff
@@ -162,7 +164,14 @@ impl State {
 	/// Determine whether an account exists.
 	pub fn exists(&self, a: &Address) -> bool {
 		let db = self.trie_factory.readonly(self.db.as_hashdb(), &self.root).expect(SEC_TRIE_DB_UNWRAP_STR);
-		self.cache.borrow().get(a).unwrap_or(&None).is_some() || db.contains(a)
+		self.cache.borrow().get(&a).unwrap_or(&None).is_some()
+		|| match db.contains(a) {
```
Comment (gavofyork) [inline ethcore/src/state.rs]: run-on line should be indented.

File: ethcore/src/state.rs [hunk 3]
```diff
@@ -248,7 +251,13 @@ impl State {
 	/// Commit accounts to SecTrieDBMut. This is similar to cpp-ethereum's dev::eth::commit.
 	/// `accounts` is mutable because we may need to commit the code or storage and record that.
 	#[cfg_attr(feature="dev", allow(match_ref_pats))]
-	pub fn commit_into(trie_factory: &TrieFactory, db: &mut HashDB, root: &mut H256, accounts: &mut HashMap<Address, Option<Account>>) {
+	pub fn commit_into(
+		trie_factory: &TrieFactory,
+		db: &mut HashDB,
+		root: &mut H256,
+		accounts: &mut HashMap<Address,
+		Option<Account>>
```
Comment (tomusdrw) [inline ethcore/src/state.rs]: Could stay on the same line.

File: ethcore/src/account.rs (hunk 1 above)
Comment (debris) [inline ethcore/src/account.rs]: or `x.map_or_else(U256::default, decode)` to avoid importing `Uint` trait :)

File: ethcore/src/state.rs (hunk 3 above)
Comment (rphmeier) [inline ethcore/src/state.rs]: it could, but the line stands at something like 130 characters plus a tab. it seems cleaner to indent it in some way.

**Comment by gavofyork** [comment]
looks generally good.

would be nice to re-review against the current master - there are a lot of unrelated merges in the diff here.

**Comment by rphmeier** [comment]
@gavofyork it is merged with the latest master -- i hadn't realized it was messing up the diff.

**Comment by gavofyork** [comment]
travis failing?

**Comment by rphmeier** [comment]
seems like it, not sure why -- all the tests are passing on my machine ;)
I wonder if it's spurious, since the test failure here is `miner::should_get_price_info`. completely unrelated to trie.

**Comment by gavofyork** [comment]
yeah - that one requires network access.

**Pull request merged**